buildpacks/nodejs-npm-install/CHANGELOG.md

```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Removed
+
+- Check and warning for `node_modules` directory is unnecessary since
+  `npm ci` will remove it before installing dependencies. ([#998]https://github.com/heroku/buildpacks-nodejs/pull/998))
```

Comment:
(not blocking) Does `npm ci` warn if it removes `node_modules`? If it doesn't, I wonder if there may end up being a non-zero amount of apps whose builds are slower / more likely to hit git limits due to forgetting to add their `node_modules` to `.gitignore`? Though I suppose if the goal is to encourage optimised apps (rather than as an npm-specific workaround), I guess the check would need to be moved to the engine buildpack anyway, so it displays the warning for yarn/pnpm apps too?

Reply:
No, there is no warning. It's implied by the usage of `npm ci` and since the output shows the command when installing, I consider that a loose form of notification.

Could some installs be slower? I hadn't considered the performance impact that would come from the deletion though having your `node_modules` directory checked into git would have a broader impact than just during dependency installation and you'd likely see other negative side-effects. 

Either way, this warning is in the wrong place here but I'm still trying to figure out how to tackle some of these application checks in a way that lets us add Heroku-specific warnings and best practices such as Node version support deprecations (like Python does) without impeding the individual functionality of some of these buildpacks.

Reply:
> Could some installs be slower?

If the user is Git pushing a 1GB `node_modules` directory rather than only the app source, I'm presuming there will be a reasonably performance impact.